mdoc/src/main/scala/mdoc/PostModifier.scala

```diff
@@ -1,19 +1,25 @@
 package mdoc
 
 import java.util.ServiceLoader
-import mdoc.internal.cli.Settings
+
+import mdoc.internal.cli.{Exit, Settings}
 import metaconfig.ConfDecoder
 import metaconfig.ConfEncoder
 import metaconfig.ConfError
 import metaconfig.generic.Surface
+
 import scala.meta.inputs.Input
 import scala.meta.io.AbsolutePath
 import scala.collection.JavaConverters._
 import scala.meta.io.RelativePath
 
 trait PostModifier {
   val name: String
+  def onStart(settings: MainSettings): Unit = ()
+  def preProcess(ctx: PostModifierContext): Unit = ()
   def process(ctx: PostModifierContext): String
+  def postProcess(ctx: PostModifierContext): Unit = ()
+  def onExit(exit: Exit): Unit = ()
```

Comment:
`Exit` is an internal API so we'll need to pass in a different type that is inside the toplevel `mdoc` package

Reply:
Will set it to `Int`. 0 - success, 1 - failure.

Reply:
It’s better to wrap it in a custom class that we can extend in the future.

Reply:
Done